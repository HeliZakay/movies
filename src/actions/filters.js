// SET_TEXT_FILTER
export const setTextFilter = (text = '') => ({
    type: 'SET_TEXT_FILTER',
    text
  });

  export const setGenresFilter = (genresArray) => ({
    type: "SET_GENRES_FILTER",
    genresArray
  });

// SORT_BY_DATE
export const sortByDate = () => ({
    type: 'SORT_BY_DATE'
  });
  
  // SORT_BY_SCORE
  export const sortByScore = () => ({
    type: 'SORT_BY_SCORE'
  });

  // SET_PERSON_FILTER
  export const setPersonFilter = (person = "") => ({
    type: "SET_PERSON_FILTER",
    person
  });

