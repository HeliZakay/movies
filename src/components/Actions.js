import React from "react";
import { Link} from 'react-router-dom';

export default () => {
    return (
        <div className="actions-section">
        <div className="content-container">
        <div className="actions-section__wrapper">
        
        <div className="actions-section__item">
            <Link className="actions-section__link" to="/create">
            <div className="actions-section__background">
            <img className="actions-section__img" src="images/popcorn.png"></img> 
            </div>
            <h3 className="actions-section__text">Add a movie</h3>
            
            </Link>
        </div>
        <div className="actions-section__item">
            <Link className="actions-section__link" to="/friends">
            <div className="actions-section__background">
            <img className="actions-section__img" src="images/familyguy.png"></img>  
            </div>
            <h3 className="actions-section__text">Add a friend</h3>
           
            </Link>
        </div>
        <div className="actions-section__item">
            <Link className="actions-section__link" to="/contact">
            <div className="actions-section__background">
            <img className="actions-section__img" src="images/envelop.png"></img>  
            </div>
            <h3 className="actions-section__text">Send feedback</h3>
            
            </Link>
        </div>
        </div>  
        </div>          
        </div>
    );
};



