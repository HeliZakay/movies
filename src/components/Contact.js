import React from "react";
import {connect} from "react-redux";

export class Contact extends React.Component {
    onSubmit = (event) => {
        event.preventDefault();
        this.props.history.push("/homepage");
    }
    render() {
        return (
            <form
            action="/" 
            method="post" 
            encType="text/plain"
            // onSubmit={this.onSubmit}
            className="form" >

            <div className="page-header">
            <div className="content-container--form">
                 <h2 className="page-header__title">Send Feedback</h2>
            </div>
        </div>
        <div className="content-container--form">
           <p>
            The goal here is to make your experience as much pleasant as possible!
           </p>
           <textarea className="textarea text-area--contact"
                        placeholder="Write your feedback here"
                        name="feedback" autoFocus>
        </textarea>
        <div>
                <button type="submit" className="button button--form"><strong>Send!</strong></button>
        </div>
        </div>
        </form>

        );
    };
}

    
 export default connect()(Contact);
  
