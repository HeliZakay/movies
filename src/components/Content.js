import React from 'react';
import {connect} from "react-redux";

export  class Content extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            readMore: false
        }
    }
    readMore = () => {
        this.setState({readMore: !this.state.readMore});
    }
    shortenContent = (content) => {
        if(content.length< 150) {
            return <p>"{content}"</p>;
        }
        else {
            return <div>
            {!this.state.readMore ?
            <p>
            "{content.slice(0,150)}..."
            <a onClick={this.readMore}>
            {this.props.language === "English"?  "Read More" : "קרא עוד "}
            </a>
            </p>:
            <p>
            "{content}"
            <a onClick={this.readMore}>
            {this.props.language === "English"?  "Read Less" : "קרא פחות "}
            </a>
            </p>
            }
            </div>;
        }
    }
    render() {
        return (
            <div className="content">
             {this.props.content && <div className="card-text"> {this.shortenContent(this.props.content)}</div>}           
            </div>
        );
    }
};



const mapStateToProps = (state) => ({
    language: state.auth.language
});

export default connect(mapStateToProps)(Content);
   