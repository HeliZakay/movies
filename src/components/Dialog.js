import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import MuiDialogTitle from '@material-ui/core/DialogTitle';
import MuiDialogContent from '@material-ui/core/DialogContent';
import MuiDialogActions from '@material-ui/core/DialogActions';
import IconButton from '@material-ui/core/IconButton';
import CloseIcon from '@material-ui/icons/Close';
import Typography from '@material-ui/core/Typography';
import {connect} from "react-redux";
import MovieCard from "./MovieCard";


const styles = (theme) => ({
  root: {
    margin: 0,
    padding: theme.spacing(2),
  },
  closeButton: {
    position: 'absolute',
    right: theme.spacing(1),
    top: theme.spacing(1),
    color: theme.palette.grey[500],
  }
});

const DialogTitle = withStyles(styles)((props) => {
  const { children, classes, onClose, ...other } = props;
  return (
    <MuiDialogTitle disableTypography className={classes.root} {...other}>
      <Typography variant="h6">{children}</Typography>
      {onClose ? (
        <IconButton aria-label="close" className={classes.closeButton} onClick={onClose}>
          <CloseIcon />
        </IconButton>
      ) : null}
    </MuiDialogTitle>
  );
});

const DialogContent = withStyles((theme) => ({
  root: {
    padding: theme.spacing(2),
  },
}))(MuiDialogContent);

export function CustomizedDialogs(props) {
  const [open, setOpen] = React.useState(false);

  const handleClickOpen = () => {
    setOpen(true);
  };
  const handleClose = () => {
    setOpen(false);
  };

  return (
    <div >
    <div className="dialog__button-div">
      <button className="btn button-friend--message btn-lg button-open-review" onClick={handleClickOpen}>
       {props.language === "English"? "Open review": "פתח ביקורת"}
      </button>
      </div>
      <Dialog 
      className={String(props.language !== "English" && "align-right")}
      onClose={handleClose} aria-labelledby="customized-dialog-title" open={open}>
        <DialogTitle id="customized-dialog-title" onClose={handleClose}>
          {props.language === "English" || !props.movie.hname?
          (props.movie.movieName ): (props.movie.hname)}
        </DialogTitle>
        <DialogContent dividers>
         
           <MovieCard 
                        className={String(props.language !== "English" && "align-right")}
                        reviews = {props.movie.reviews}
                        id={props.movie.id}
                        movieName={props.movie.movieName}
                        hname={props.movie.hname}
                        imdbData={props.movie.imdbData}
                        dialog={true}
                     />
        </DialogContent> 
      </Dialog>
    </div>
  );
}

const mapStateToProps = (state) => ({
  language: state.auth.language,
  movies: state.movies
});
export default connect(mapStateToProps)(CustomizedDialogs);