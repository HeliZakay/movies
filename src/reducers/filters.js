// Filters Reducer
const filtersReducerDefaultState = {
    text: '',
    person: '',
    sortBy: 'date',
  };
  
export default (state = filtersReducerDefaultState, action) => {
    switch (action.type) {
      case 'SET_TEXT_FILTER':
        return {
          ...state,
          text: action.text
        };
      case "SET_PERSON_FILTER":
        return {
          ...state,
          person: action.person
        }
      case 'SORT_BY_SCORE':
        return {
          ...state,
          sortBy: 'score'
        };
      case 'SORT_BY_DATE':
        return {
          ...state,
          sortBy: 'date'
        };
      default:
        return state;
    }
  };
