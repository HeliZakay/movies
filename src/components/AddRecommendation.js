import React from "react";
import RecommendationForm from "./RecommendationForm";
import {connect} from "react-redux";
import {startAddMovie} from "../actions/movies";

export class AddRecommendation extends React.Component {
   onSubmit = (movie) => {
      this.props.startAddMovie(movie);
      this.props.history.push("/");
   };
   render() {
      return (
         <div>
         <h1>Add Recommendation</h1>
         <RecommendationForm
           onSubmit={this.onSubmit}
          />
         </div>
      );
   }
}


const mapDispatchToProps = (dispatch) => ({
   startAddMovie: (movie) => dispatch(startAddMovie(movie))
});
   
export default connect(undefined, mapDispatchToProps)(AddRecommendation);

// import React from 'react';
// import { connect } from 'react-redux';
// import RecommendationForm from './RecommendationForm';
// import { startAddMovie } from '../actions/movies';

// export class AddMovie extends React.Component {
//   onSubmit = (movie) => {
//     this.props.startAddMovie(movie);
//     this.props.history.push('/');
//   };
//   render() {
//     return (
//       <div>
//         <h1>Add Movie</h1>
//         <RecommendationForm
//           onSubmit={this.onSubmit}
//         />
//       </div>
//     );
//   }
// }

// const mapDispatchToProps = (dispatch) => ({
//   startAddMovie: (movie) => dispatch(startAddMovie(movie))
// });

// export default connect(undefined, mapDispatchToProps)(AddMovie);