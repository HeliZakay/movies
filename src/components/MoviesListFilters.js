import React from "react";
import {connect} from "react-redux";
import  {setPersonFilter, sortByDate, sortByScore, setTextFilter} from "../actions/filters";

export class MoviesListFilters extends React.Component {
    onMovieChange = (event) => {
        this.props.setTextFilter(event.target.value);
    };
    onPersonChange = (event) => {
        this.props.setPersonFilter(event.target.value);
    };
    onSortChange = (event) => {
        
        event.target.value === "date"?
        (this.props.sortByDate())
        : (this.props.sortByScore())
        
    };
    render() {
        return (
            <div className="content-container filters">
                <div className="input-group">
                    <div className="input-group__item">
                    <input className="text-input" 
                    type="text" 
                    placeholder={this.props.language === "English"? "filter by movie name": "סנן לפי שם סרט"}
                    value={this.props.filters.text}
                    onChange={this.onMovieChange}    
                    />
                    </div>

                    <div className="input-group__item">
                    <input className="text-input"
                    type="text" 
                     placeholder={this.props.language === "English"? "filter by recommender": "סנן לפי שם הממליץ"}
                     value={this.props.filters.person}
                      onChange={this.onPersonChange}    
                     />
                    </div>
                    
                    <div className="input-group__item">
                    <select className="select"
                    value={this.props.filters.sortBy}
                     onChange ={this.onSortChange} 
                    >
                         <option value="date">{this.props.language === "English"? "Latest": "הכי חדשים"}</option>
                         <option value="score">{this.props.language === "English"? "Top Rated": "הניקוד הגבוה ביותר"}</option>
                     </select>
                    </div>
                </div>
        </div>
        );
    }
};

const mapStateToProps = (state) => ({
    filters: state.filters,
    language: state.auth.language
});

const mapDispatchToProps = (dispatch) => ({
    setTextFilter: (movieName) => dispatch(setTextFilter(movieName)),
    setPersonFilter: (person) => dispatch(setPersonFilter(person)),
    sortByDate: () => dispatch(sortByDate()),
    sortByScore: () => dispatch(sortByScore())
});


export default connect(mapStateToProps, mapDispatchToProps)(MoviesListFilters);