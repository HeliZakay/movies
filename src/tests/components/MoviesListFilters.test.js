import React from "react";
import {shallow} from "enzyme";
import {MoviesListFilters} from "../../components/MoviesListFilters";
import {filters,altFilters} from "../fixtures/filters";

let setTextFilter, setPersonFilter, sortByDate, sortByScore, wrapper;

beforeEach(() => {
    setTextFilter = jest.fn();
    setPersonFilter = jest.fn();
    sortByScore = jest.fn();
    sortByDate = jest.fn();
    wrapper = shallow(
        <MoviesListFilters 
        filters={filters}
        setTextFilter={setTextFilter}
        setPersonFilter={setPersonFilter}
        sortByDate={sortByDate}
        sortByScore={sortByScore}
        />
        );
});

test("should render MoviesListFilters correctly", () => {
    expect(wrapper).toMatchSnapshot();
});


test("should render MoviesListFilters with alt data correctly", () => {
    wrapper.setProps({
        filters: altFilters
    });
    expect(wrapper).toMatchSnapshot();
});

test("should handle text change", () => {
    const value = "tarazan";
    wrapper.find("input").at(0).simulate("change", {
        target: {value}
    });
    expect(setTextFilter).toHaveBeenLastCalledWith(value);
});

test("should handle person change", () => {
    const value = "heli";
    wrapper.find("input").at(1).simulate("change", {
        target: {value}
    });
    expect(setPersonFilter).toHaveBeenLastCalledWith(value);
});


test("should sort by date ", () => {
    const value = "date";
    wrapper.setProps({
        filters: altFilters
    });
    wrapper.find("select").simulate("change", {
        target: {value}
    });
    expect(sortByDate).toHaveBeenCalled();
});



test("should sort by score ", () => {
    const value = "score";
    wrapper.find("select").simulate("change", {
        target: {value}
    });
    expect(sortByScore).toHaveBeenCalled();
});


