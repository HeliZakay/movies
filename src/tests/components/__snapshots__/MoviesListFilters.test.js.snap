// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MoviesListFilters correctly 1`] = `
<div>
  <input
    onChange={[Function]}
    placeholder="filter by movie name"
    type="text"
    value=""
  />
  <input
    onChange={[Function]}
    placeholder="filter by recommender"
    type="text"
    value=""
  />
  <select
    onChange={[Function]}
    value="date"
  >
    <option
      value="date"
    >
      Latest
    </option>
    <option
      value="score"
    >
      Top Rated
    </option>
  </select>
</div>
`;

exports[`should render MoviesListFilters with alt data correctly 1`] = `
<div>
  <input
    onChange={[Function]}
    placeholder="filter by movie name"
    type="text"
    value="tarazan"
  />
  <input
    onChange={[Function]}
    placeholder="filter by recommender"
    type="text"
    value="heli"
  />
  <select
    onChange={[Function]}
    value="score"
  >
    <option
      value="date"
    >
      Latest
    </option>
    <option
      value="score"
    >
      Top Rated
    </option>
  </select>
</div>
`;
